### DISCUSSION TITLE
Transformations for 3d images

### DISCUSSION THREAD

**Discussion by jakubMitura14** [deepmind/dm_pix#60] [Q&A] [answered]
Hello 2 questions 
1) as the transformations are implemented in pure Jax they are differentiable in most cases - am I right? (more precisely I am considering putting the probability of applying transform as learnable parameter)
2) is it possible to use transformations in 3D images (Magnetic resonance imagiing)?

**Comment by pabloduque0** [comment]
Hello @jakubMitura14 !

1. That will be transformation dependent, I know some are differentiable but I would not be able to answer about all of them as a whole. If you have questions about specific ones, we can take a look at those. 
2. What you can do is `vmap` the transformations you want to do over the slice (slice as in MRI slice) axis. That way you can apply the same transformation to the whole 3D image. However that will still apply 2D transformations. There might be some of the transformations that are 3D friendly but from what I recall most actually expect the third dimension to be channel and treat it as such.

**Comment by jakubMitura14** [comment]
Thank You ! I am mainly thinking about the differentiability of brightness, gamma, rotate and elastic deformation

**Comment by claudiofantacci** [comment, marked as answer]
Hey @jakubMitura14, they are all differentiable indeed. If you hit any issues with your setup/experiment, please do let us know so we can look into it (or you can have a look and send a PR if you want to!) 😄
